src/Nethermind/Nethermind.JsonRpc.Test/Modules/Eth/EthRpcModuleTests.cs

```diff
@@ -1161,13 +1195,36 @@ public async Task Eth_create_access_list_sample(AccessListProvided accessListPro
     [TestCase(0)]
     public static void Should_handle_gasCap_as_max_if_null_or_zero(long? gasCap)
     {
-        TransactionForRpc rpcTx = new TransactionForRpc();
+        LegacyTransactionForRpc rpcTx = new LegacyTransactionForRpc();
 
         rpcTx.EnsureDefaults(gasCap);
 
         Assert.That(rpcTx.Gas, Is.EqualTo(long.MaxValue), "Gas must be set to max if gasCap is null or 0");
     }
 
+    [Ignore(reason: "Shows disparity across 'default' methods")]
+    [TestCase(null)]
+    [TestCase(0)]
+    public static void ToTransactionWithDefaults_and_EnsureDefaults_same_GasLimit(long? gasCap)
+    {
+        long toTransactionWitDefaultsGasLimit;
+        {
+            var rpcTx = new LegacyTransactionForRpc();
+            Transaction tx = rpcTx.ToTransaction();
+            toTransactionWitDefaultsGasLimit = tx.GasLimit;
+        }
+
+        long ensureDefaultsGasLimit;
+        {
+            var rpcTx = new LegacyTransactionForRpc();
+            rpcTx.EnsureDefaults(gasCap);
+            var tx = rpcTx.ToTransaction();
+            ensureDefaultsGasLimit = tx.GasLimit;
+        }
+
+        toTransactionWitDefaultsGasLimit.Should().Be(ensureDefaultsGasLimit);
+    }
```

Comment:
It's not clear to me why we have the `EnsureDefault`s method. Again, can we delete it?